rust/src/table_state.rs

```diff
@@ -211,6 +222,11 @@ impl DeltaTableState {
         if !new_state.commit_infos.is_empty() {
             self.commit_infos.append(&mut new_state.commit_infos);
         }
+
+        // TODO should we make sure versions are consistent (i.e. +1) or just update to highest?
+        if self.version < new_state.version {
+            self.version = new_state.version
+        }
```

Comment:
Here i was unsure how to best proceed. updating the version on merge seems a bit iffy to me. At best we get the state for whatever state has the highest version. However we have no way of knowing at this point if this represents an existing state, or some future state, or ... So I did not want to put any logic here.

Reply:
I'm not sure I understand the `merge` command. If you have two `DeltaTableState`s, isn't the correct one simply the newest one? Or is it possible for `DeltaTableState` to represent just incremental changes in table?

Reply:
yes, `DetaTableState` can be created from a collection of actions - so a single commit or the entire reconciled table state. so the merge function is mainly used to increment the state after updates etc... previously it had no notion of versions and is e.g. also used to reconcile all actions for a checkpoint...

Reply:
I think this behavior should be fine as long as we document it in the method documentation.

Reply:
added some more verbose doc string explaining how merge treats version numbers.